src/dodal/devices/i22/nxsas.py

```diff
@@ -1,12 +1,32 @@
+import asyncio
+from collections.abc import Awaitable, Iterable
 from dataclasses import dataclass, fields
+from typing import TypeVar
 
 from bluesky.protocols import Reading
 from event_model.documents.event_descriptor import DataKey
-from ophyd_async.core import DirectoryProvider, merge_gathered_dicts
-from ophyd_async.epics.areadetector import AravisDetector, PilatusDetector
-from ophyd_async.epics.areadetector.aravis import AravisController
+from ophyd_async.core import PathProvider
+from ophyd_async.epics.adaravis import AravisController, AravisDetector
+from ophyd_async.epics.adpilatus import PilatusDetector
 
 ValueAndUnits = tuple[float, str]
+T = TypeVar("T")
+
+
+# Temporarily duplicated non-public method from ophyd_async
```

Comment:
Because the handling of metadata as soft signals is temporary I was considering #595 in dodal, not an issue in ophyd-async, to be the solution: we don't really want to expose _merge_gathered_dicts, just use the behaviour of nested devices.

Reply:
Sure, I don't mind what the solution is, I would just like the comment to link to the issue giving more detail. I think it makes the comment less likely to rot.